lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -130,6 +130,8 @@ def success_document
           hash[:links].update(pagination_links_for(serializer))
         end
 
+        hash[:meta] = instance_options[:meta] if instance_options[:meta].is_a?(Hash)
```

Comment:
Actually we kinda want to ensure the value of `meta` is a hash (as per the spec).